### PR TITLE
DynamicDepositOrder

### PR DESCRIPTION
**Pull request by hsjhita**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ECM-77

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java [hunk 1]
```diff
@@ -365,46 +365,57 @@ public void validateJudgementsHasJurisdiction(CaseData caseData, boolean partOfM
...
-                            .getValue().getCode();
-                    if (dynamicOrderAgainst.startsWith("R:")) {
-                        depositTypeItem.getValue().setDepositOrderAgainst(RESPONDENT_TITLE);
-                    } else if (dynamicOrderAgainst.startsWith("C:")) {
-                        depositTypeItem.getValue().setDepositOrderAgainst(CLAIMANT_TITLE);
-                    } else {
-                        errors.add(UNABLE_TO_FIND_PARTY);
-                    }
-                }
+                validateDepositOrderAgainst(errors, depositTypeItem);
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java:368]: What about having a DepositOrderValidationService instead?

Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationService.java]: Refactored

**Review by Angela-LW — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DepositOrderValidationService.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+                validateDepositOrderAgainst(errors, depositTypeItem);
+                validateDepositRequestedBy(errors, depositTypeItem);
+                validateDepositRefunded(errors, depositTypeItem);
+            }
+        }
+        return errors;
+    }
+
+    public static void validateDepositRefunded(List<String> errors, DepositTypeItem depositTypeItem) {
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DepositOrderValidationService.java:41]: Why is this method and the following methods public?

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DepositOrderValidationService.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import static uk.gov.hmcts.ecm.common.model.helper.Constants.UNABLE_TO_FIND_PARTY;
+
+public class DepositOrderValidationService {
+
+    private DepositOrderValidationService() {
+    }
+
+    public static List<String> validateDepositOrder(CaseData caseData) {
+        List<String> errors = new ArrayList<>();
+        if (!CollectionUtils.isEmpty(caseData.getDepositCollection())) {
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DepositOrderValidationService.java:23]: CollectionUtils.isNotEmpty

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DepositOrderValidationService.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+            for (DepositTypeItem depositTypeItem : caseData.getDepositCollection()) {
+                if (!isNullOrEmpty(depositTypeItem.getValue().getDepositAmountRefunded())
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DepositOrderValidationService.java:26]: Put this in a method to follow the pattern below it. Also, the method name should clearly express what it is validating as the if statement is quite hard to follow

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationServiceTest.java [hunk 5]
```diff
@@ -529,22 +528,22 @@ void shouldValidateJurisdictionCodesWithinJudgementEmptyJurCodesCollection() {
 
     @Test
     void shouldValidateDepositRefunded() {
-        List<String> errors = eventValidationService.validateDepositOrder(caseDetails3.getCaseData());
+        List<String> errors = DepositOrderValidationService.validateDepositOrder(caseDetails3.getCaseData());
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationServiceTest.java:531]: You're testing the DepositOrderValidationService class in the EventValidationServiceTest.

**Review by jthmcts — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DepositOrderValidationService.java (hunk 2 above)
Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DepositOrderValidationService.java]: Refactored

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DepositOrderValidationService.java (hunk 4 above)
Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DepositOrderValidationService.java]: Extracted

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DepositOrderValidationService.java (hunk 3 above)
Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DepositOrderValidationService.java:22]: Refactored

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationServiceTest.java (hunk 5 above)
Comment (hsjhita) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/EventValidationServiceTest.java]: Created new Test Class

**Review by jthmcts — APPROVED** [review]
